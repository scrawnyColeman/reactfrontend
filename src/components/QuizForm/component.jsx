import React from 'react';
import styled from 'styled-components';
import QuizQuestionForm from '../QuizQuestionForm/component';

const QuizContainer = styled.div`
    padding: 16px;
`;
const StyledHead = styled.div`
    text-align: left;
    font-size: 2rem;
`;
const StyledBody = styled.div`
    height: 50vh;
    overflow-y: scroll;
`;
const QuizForm = () => {
    const quizQuestions = new Array(10).fill('');
    return (
        <QuizContainer>
            <StyledHead>Quiz Questions</StyledHead>
            <StyledBody>
                {quizQuestions.map((value, index) => (
                    <QuizQuestionForm key={`${value}${index}`} questionNumber={index + 1} />
                ))}
            </StyledBody>
        </QuizContainer>
    );
};

export default QuizForm;
