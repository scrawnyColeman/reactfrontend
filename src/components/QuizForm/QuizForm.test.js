import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import React from 'react';
import QuizForm from './component';

Enzyme.configure({ adapter: new Adapter() });

describe('QuizForm component', () => {
    test('component matches snapshot', () => {
        const wrapper = shallow(<QuizForm />);
        expect(wrapper).toMatchSnapshot();
    });
});
