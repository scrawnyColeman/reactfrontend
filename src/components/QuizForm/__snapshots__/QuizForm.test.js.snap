// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizForm component component matches snapshot 1`] = `
<styled.div>
  <styled.div>
    Quiz Questions
  </styled.div>
  <styled.div>
    <QuizQuestionForm
      questionNumber={1}
      setQuestionNumber={[Function]}
    />
  </styled.div>
</styled.div>
`;
